onvm/go.sh

```diff
@@ -1,37 +1,77 @@
         echo -e "\t\tRX thread, TX thread, ..., TX thread for last NF, Stats thread"
-        echo -e "$0 0,1,2,3 3 0xF0 -s web"
+        echo -e "$0 -k 3 -n 0xF0 -m 2,3,4"
+        echo -e "\tRuns ONVM the same way as above, but manually configures cores 2, 3 and 4 to be used as stated"
+        echo -e "$0 -k 3 -n 0xF0 -m 2,3,4 -s web"
         echo -e "\tRuns ONVM the same way as above, but prints statistics to the web browser"
-        echo -e "$0 0,1,2,3 3 0xF0 -s web -p 9000"
+        echo -e "$0 -k 3 -n 0xF0 -m 2,3,4 -s web -p 9000"
         echo -e "\tRuns OVNM the same as above, but runs the web stats on port 9000 instead of defaulting to 8080"
-        echo -e "$0 0,1,2,3 3 0xF0 -s stdout"
+        echo -e "$0 -k 3 -n 0xF0 -m 2,3,4 -s stdout"
         echo -e "\tRuns ONVM the same way as above, but prints statistics to stdout"
-        echo -e "$0 0,1,2,3 3 0xF0 -s stdout -c"
+        echo -e "$0 -k 3 -n 0xF0 -m 2,3,4 -s stdout -c"
         echo -e "\tRuns ONVM the same way as above, but enables shared cpu support"
-        echo -e "$0 0,1,2,3 3 0xF0 -s stdout -t 42"
+        echo -e "$0 -k 3 -n 0xF0 -m 2,3,4 -s stdout -t 42"
         echo -e "\tRuns ONVM the same way as above, but shuts down after 42 seconds"
-        echo -e "$0 0,1,2,3 3 0xF0 -s stdout -l 64"
+        echo -e "$0 -k 3 -n 0xF0 -m 2,3,4 -s stdout -l 64"
         echo -e "\tRuns ONVM the same way as above, but shuts down after receiving 64 million packets"
-        echo -e "$0 0,1,2,3 3 0xF0 -s stdout -v"
+        echo -e "$0 -k 3 -n 0xF0 -m 2,3,4 -s stdout -v"
         echo -e "\tRuns ONVM the same way as above, but prints statistics to stdout in extra verbose mode"
-        echo -e "$0 0,1,2,3 3 0xF0 -s stdout -vv"
+        echo -e "$0 -k 3 -n 0xF0 -m 2,3,4 -s stdout -vv"
         echo -e "\tRuns ONVM the same way as above, but prints statistics to stdout in raw data dump mode"
-        echo -e "$0 0,1,2,3 3 0xF0 -a 0x7f000000000 -s stdout"
+        echo -e "$0 -k 3 -n 0xF0 -m 2,3,4 -a 0x7f000000000 -s stdout"
         echo -e "\tRuns ONVM the same way as above, but adds a --base-virtaddr dpdk parameter to overwrite default address"
-        echo -e "$0 0,1,2,3 3 0xF0 -r 10 -d 2"
+        echo -e "$0 -k 3 -n 0xF0 -m 2,3,4 -r 10 -d 2"
         echo -e "\tRuns ONVM the same way as above, but limits max service IDs to 10 and uses service ID 2 as the default"
         exit 1
 }
 
-cpu=$1
-ports=$2
-nf_cores=$3
+# User can still use legacy syntax for backwards compatibility. Check syntax of input
+# Check validity of core input
+core_check="^([0-8]+,){2}([0-8]+)(,[0-8]+)*$"
+port_check="^[0-9]+$"
+nf_check="^0x[0-9A-F]+$"
+flag_check="^-[ardstlpzcvmkn]$"
+if [[ $1 =~ $core_check ]] && [[ $2 =~ $port_check ]] && [[ $3 =~ $nf_check ]]
+then
+    cpu=$1
+    ports=$2
+    nf_cores=$3
+    shift 3
+# Make sure someone isn't inputting the cores incorrectly and they are using legacy syntax
+elif [[ ! $1 =~ $flag_check ]]
+then
+    if [[ ! $1 =~ $core_check ]] 
```

Comment:
Can we make these into variables? We check them quite often. Why don't we test in the beginning, create a boolean and clean it up slightly? ie `$1 =~ $core_check`

Reply:
I think I figured out how to do this. It's in my most recent commit.